thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionTaskRunner.java

```diff
@@ -65,6 +65,7 @@
   private List<String> collectionDimensions;
   private AnomalyFunctionFactory anomalyFunctionFactory;
   private BaseAnomalyFunction anomalyFunction;
+  private BaseAnomalyFunction ruleBasedAnomalyFunction;
```

Comment:
I wonder if it's better to call this "secondary" instead of "ruleBased". After all, any anomaly function should work (minus collisions in the properties)

Reply:
why not a list of anomalyFunctions?